ruby/vampires.rb

```diff
@@ -53,13 +49,23 @@
   wants_insurance = false
 end
 
+allergies = false
+until allergies == "done" || allergies == "sunshine"
 puts "any allergies? type done if no"
 allergies = gets.chomp
-until allergies == "done" || allergies == "sunshine"
-  allergies = gets.chomp
+  if allergies == "sunshine"
+    puts "Hmm."
+  elsif allergies == "done"
+    puts "Ok, great."
+  else
+    puts "I didn't understand you"
+  end
 end
 
-if got_age_right == 0 && likes_garlic_bread && wants_insurance && name = true
+if name != true || allergies == "sunshine"
+  puts "Definitely a vampire"
+
+elsif got_age_right == 0 && likes_garlic_bread && wants_insurance && name = true
   puts "Probably not a vampire"
 
 elsif got_age_right >= 1 && likes_garlic_bread != true && wants_insurance != true
```

Comment:
Ruby golf refactor (golf == shorter lines for design)

Since some of your variables contain booleans you can just return those values to evaluate.

Read this update carefully.  It does the same thing :)

`got_age_right >= 1 && likes_garlic_bread != true && wants_insurance != true`

can be

`got_age_right >= 1 && !likes_garlic_bread && !wants_insurance`

another example:

`got_age_right == 0 && likes_garlic_bread && wants_insurance && name = true`

can be

`got_age_right == 0 && likes_garlic_bread && wants_insurance && name`